src/foam/dao/SyncDAO.js

```diff
@@ -112,6 +118,18 @@ foam.CLASS({
       factory: function() { return Promise.resolve(); }
     },
     {
+      name: 'versionedOf_',
+      expression: function(of) {
+        return this.versionedClassFactory_.get(of);
+      }
```

Comment:
I agree. At the moment, I'm using this `SyncDAO` implementation (so I need to land it), but I do not depend on `SyncDAO` owning this adaptation (current usage is `of: theVersionedClass` in server <--> server interactions).

All that is to say, pulling this out to `TypeAdapterDAO` should be a prerequisite to integrating with `EasyDAO`, but I don't have time to implement it right now.